index.d.ts

```diff
@@ -0,0 +1,438 @@
+
+/**
+ * Describes the state that the task instance is in.
+ * Can be used for debugging, or potentially driving some UI state.
+ */
+export enum TaskInstanceState {
+  /**
+   * Task instance was canceled before it started.
+   */
+  Dropped = 'dropped',
+
+  /**
+   * Task instance was canceled before it could finish.
+   */
+  Canceled = 'canceled',
+
+  /**
+   * Task instance ran to completion, even if an exception was thrown.
+   */
+  Finished = 'finished',
+
+  /**
+   * Task instance is currently running. Returns `true`, even if it is paused
+   * on a yielded promise.
+   */
+  Running = 'running',
+
+  /**
+   * Task instance hasn't begun running yet. Usually because the task is using
+   * the `.enqueue()` task modifier.
+   */
+  Waiting = 'waiting'
+}
+
+/**
+ * A `TaskInstance` represents a single execution of a `Task`. Every call to
+ * `Task#perform` returns a `TaskInstance`.
+ *
+ * `TaskInstance`s are cancelable, either explicitly via `TaskInstance#cancel`
+ * or `Task#cancelAll`, or automatically due to the host object being
+ * destroyed, or because concurrency policy enforced by a Task Modifier
+ * canceled the task instance.
+ */
+interface TaskInstanceBase<T> extends PromiseLike<T>, Getter {
+  /**
+   * Describes the state that the task instance is in.
+   * Can be used for debugging, or potentially driving some UI state.
+   */
+  // readonly state: TaskInstanceState;
+
+  /**
+   * `true` if the task instance has started, else `false`.
+   */
+  readonly hasStarted: boolean;
+
+  /**
+   * `true` if the task instance was canceled before it could run to
+   * completion.
+   */
+  readonly isCanceled: boolean;
+
+  /**
+   * `true` if the `TaskInstance` was canceled before it could ever start
+   * running.
+   *
+   * For example, calling `.perform()` twice on a task with the `.drop()`
+   * modifier applied will result in the second task instance being dropped.
+   */
+  readonly isDropped: boolean;
+
+  /**
+   * `true` if the task has run to completion.
+   */
+  readonly isFinished: boolean;
+
+  /**
+   * `true` if the task is still running.
+   */
+  readonly isRunning: boolean;
+
+  /**
+   * `true` if the task instance is fulfilled.
+   */
+  readonly isSuccessful: boolean;
+
+  /**
+   * If this `TaskInstance` runs to completion by returning a value other than
+   * a rejecting promise, this property will be set with that value.
+   */
+  readonly value?: T;
+
+  /**
+   * If this `TaskInstance` is canceled or throws an error (or yields a
+   * promise that rejects), this property will be set with that error.
+   *
+   * Otherwise, it is `null`.
+   */
+  readonly error?: Error;
+
+  /**
+   * Cancels the task instance. Has no effect if the task instance has already
+   * been canceled or has already finished running.
+   */
+  cancel(): void;
+
+  catch(): RSVP.Promise<unknown>;
```

Comment:
Are there any downsides to using the native Promise here? 

```suggestion
  catch(): Promise<unknown>;
```

Reply:
Yes: it’s not what the library actually returns! RSVP is a compatible superset type-wise, so you can test it as a normal `Promise` in your own codebase as a consumer, but if you *need* to take advantage of its being RSVP, you can’t if you make this change.